contracts/LiquidityBridgeContract.sol

```diff
@@ -11,535 +11,522 @@ import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+    return isRegistered(addr) && collateral[addr] >= minCollateral;
+  }
 
-    /**
+  /**
         @dev Checks whether a liquidity provider can deliver a pegout service
         @return Whether the liquidity provider is registered and has enough locked collateral
      */
-    function isOperationalForPegout(address addr) external view returns (bool) {
-        isRegisteredForPegout(addr) && pegoutCollateral[addr] >= minCollateral;
-    }
+  function isOperationalForPegout(address addr) external view returns (bool) {
+    isRegisteredForPegout(addr) && pegoutCollateral[addr] >= minCollateral;
+  }
 
-    /**
+  /**
         @dev Registers msg.sender as a liquidity provider with msg.value as collateral
      */
-    function register(
-        string memory _name,
-        uint _fee,
-        uint _quoteExpiration,
-        uint _acceptedQuoteExpiration,
-        uint _minTransactionValue,
-        uint _maxTransactionValue,
-        string memory _apiBaseUrl,
-        bool _status
-    ) external payable onlyEoa returns (uint) {
-        //require(collateral[msg.sender] == 0, "Already registered");
-        validateRegisterParameters(
-            _name,
-            _fee,
-            _quoteExpiration,
-            _acceptedQuoteExpiration,
-            _minTransactionValue,
-            _maxTransactionValue,
-            _apiBaseUrl
-        );
-        // TODO multiplication by 2 is a temporal fix until we define solution with product team
-        require(msg.value >= minCollateral * 2, "Not enough collateral");
-        require(
-            resignationBlockNum[msg.sender] == 0,
-            "Withdraw collateral first"
-        );
-        // TODO split 50/50 between pegin and pegout is a temporal fix until we define solution with product team
-        if (msg.value % 2 == 0) {
-            collateral[msg.sender] = msg.value / 2;
-            pegoutCollateral[msg.sender] = msg.value / 2;
-        } else {
-            collateral[msg.sender] = msg.value / 2 + 1;
-            pegoutCollateral[msg.sender] = msg.value / 2;
-        }
-
-        providerId++;
-        liquidityProviders[providerId] = LiquidityProvider({
-            id: providerId,
-            provider: msg.sender,
-            name: _name,
-            fee: _fee,
-            quoteExpiration: _quoteExpiration,
-            acceptedQuoteExpiration: _acceptedQuoteExpiration,
-            minTransactionValue: _minTransactionValue,
-            maxTransactionValue: _maxTransactionValue,
-            apiBaseUrl: _apiBaseUrl,
-            status: _status
-        });
-        emit Register(providerId, msg.sender, msg.value);
-        return (providerId);
+  function register(
+    string memory _name,
+    uint _fee,
+    uint _quoteExpiration,
+    uint _acceptedQuoteExpiration,
+    uint _minTransactionValue,
+    uint _maxTransactionValue,
+    string memory _apiBaseUrl,
+    bool _status,
+    string memory _providerType
```

Comment:
maybe its better to add this field as an enum with {PEGIN, PEGOUT, BOTH} because if its a normal string any typo would generate a 4th type of provider

Reply:
Thanks, but instead of enum found that string comparison would be more straightforward, since we would still have to compare the enum types to strings later. Updating the PR now.
`require(
      keccak256(abi.encodePacked(_providerType)) ==
        keccak256(abi.encodePacked("pegin")) ||
        keccak256(abi.encodePacked(_providerType)) ==
        keccak256(abi.encodePacked("pegout")) ||
        keccak256(abi.encodePacked(_providerType)) ==
        keccak256(abi.encodePacked("both")),
      "Invalid provider type"
    );`